modules/gapi/src/backends/fluid/gfluidcore_func.simd.hpp

```diff
@@ -1536,6 +1583,355 @@ MULC_SIMD(float, float)
 
 #undef MULC_SIMD
 
+//-------------------------------------------------------------------------------------------------
+
+template<typename scale_tag_t, typename SRC, typename DST>
+CV_ALWAYS_INLINE
+typename std::enable_if<(std::is_same<DST, ushort>::value ||
+                         std::is_same<DST, short>::value), int>::type
+divc_simd_common_impl(scale_tag_t s_tag, const SRC* inx, DST* outx,
+                      const v_float32& v_scalar, const v_float32& v_scale,
+                      const int length)
+{
+    constexpr int nlanes = vector_type_of_t<DST>::nlanes;
+
+    v_float32 v_zero = vx_setzero_f32();
+    v_float32 v_mask = (v_scalar == v_zero);
+
+    int x = 0;
+    for (;;)
+    {
+        for (; x <= length - nlanes; x += nlanes)
+        {
+            v_float32 a1 = vg_load_f32(inx);
+            v_float32 a2 = vg_load_f32(&inx[nlanes/2]);
+
+            v_store_i16(outx, v_round(v_select(v_mask, v_zero, div_op(s_tag, a1, v_scalar, v_scale))),
+                              v_round(v_select(v_mask, v_zero, div_op(s_tag, a2, v_scalar, v_scale))));
+        }
+
+        if (x < length)
+        {
+            x = length - nlanes;
+            continue;  // process unaligned tail
```

Comment:
Do we expect that `inx` and `outx` may be the same buffer actully? I means `in-place` operation

Reply:
G-API Fluid is not support in-place operations.